src/containers/JobSearch/index.tsx

```diff
@@ -1,11 +1,40 @@
-import { ReactElement } from "react";
-import { useSelector } from "react-redux";
+import React, { ReactElement, useRef, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import { Link } from "react-router-dom";
 import { animated, useSpring } from "react-spring";
+import { fetchJobs } from "../redux/actions/jobActions";
 import { RootStore } from "../redux/store";
 import "./jobSearch.scss";
 
 const JobSearch: React.FC = () => {
+  //Get Data Jobs
+  const jobs = useSelector((state: RootStore) => state.allJobs);
+
+  const dispatch = useDispatch();
+  //Initialize the variable to receive text and set text
+  const [searchJobByTitle, setSearchJobByTitle] = useState<string>("");
+
+  //Create Variable Default with Hooks useRef - Debounce
+  const typingTimeOutRef = useRef<ReturnType<typeof setInterval> | null>(null);
+
+  //Initialize function to receive text on change
+  const handleChangeWithDebounce = (e: React.ChangeEvent<HTMLInputElement>) => {
+    //setTimeout -- ClearTimeout Old -> Timeout News
+    //Exp: setTimeout 100ms --Clear -> Timeout News :300ms
+    if (typingTimeOutRef.current) {
+      clearTimeout(typingTimeOutRef.current);
+    }
+    const valueTarget: string = e.target.value;
+
+    //setTimeout when after {300 or another time period} milliseconds no input will automatically send
+    typingTimeOutRef.current = setTimeout(() => {
+      return setSearchJobByTitle(valueTarget);
+    }, 300);
+  };
```

Comment:
Debounce search just using for real-time data loading. Currently, we don't need this.